python/mxnet/gluon/trainer.py

```diff
@@ -49,6 +49,10 @@ class Trainer(object):
         on the type of compression being used. For example, 2bit compression requires a threshold.
         Arguments would then be {'type':'2bit', 'threshold':0.5}
         See mxnet.KVStore.set_gradient_compression method for more details on gradient compression.
+    update_on_kvstore : bool, default None
+        Whether to perform parameter updates on kvstore. If None, then trainer will choose the more
+        suitable option depending on the type of kvstore. If True while not using kvstore, this
```

Comment:
```
If True while not using kvstore, this option is ignored.
```
I think it's more appropriate to throw a warning when this happens. The user is not aware if MXNet decides to create a kvstore or not